README.md
# web-github-issue-reporting
Simple (KISS) user friendly webpage to report a GitHub issue,
with DB backup (SQlite, not yet).

- Responsive with Bootstrap 4.
- Antispam
- Copy to web server, edit config vars, and run.  

## Getting Started

These instructions will get you a copy of the project up and running on your local machine for development and testing purposes. See deployment for notes on how to deploy the project on a live system.

### Prerequisites

What things you need to install the software and how to install them

```
PHP >= 5.6
Sqlite 3
Google Recaptcha v3 Keys
Github Project
```

### Installing

A step by step series of examples that tell you how to get a development env running

1. Copy the folder to subdirectory/subdomain.
2. Go to Google and register the website to get [Recaptcha](https://www.google.com/recaptcha/admin#list)
3. Config de app:

```
3.1 put Google Recaptcha v3 keys in config.php
3.2 in header of index.php set Github user data:
$GithubUsernameToken = "";
$GithubUsername = "JavierCanon";
$GithubRepository = "web-github-issue-reporting";
```

#### Screenshots

![](docs/img/screenshot-localhost-37944-2019.01.17-18-42-05.png?raw=true)

## Running the tests

Explain how to run the automated tests for this system

### Break down into end to end tests

Explain what these tests test and why

```
Give an example
```

### And coding style tests

Explain what these tests test and why

```
Give an example
```

## Deployment

Add additional notes about how to deploy this on a live system

## Built With

* [PHP Tools for Visual Studio]() - The PHP dev web tool used


## Contributing

Please read [CONTRIBUTING.md](https://gist.github.com/PurpleBooth/b24679402957c63ec426) for details on our code of conduct, and the process for submitting pull requests to us.

## Versioning

We use [SemVer](http://semver.org/) for versioning. For the versions available, see the [tags on this repository](https://github.com/your/project/tags). 

## Authors

* **Javier Cañon** - *Initial work* - [JavierCanon](https://www.javiercanon.com)

See also the list of [contributors](https://github.com/your/project/contributors) who participated in this project.

## License

This project is licensed under the MIT License - see the [LICENSE.md](LICENSE.md) file for details

## Acknowledgments

* Hat tip to anyone whose code was used
* Inspiration
* etc



CHANGELOG.md
# Changelog

## 0.0.1 - 2014-05-31
### Added
- This CHANGELOG file to hopefully serve as an evolving example of a
  standardized open source project CHANGELOG.


###### Notes
The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

# Example of use:

## [Unreleased]

## [1.0.0] - 2017-06-20
### Added
- New visual identity by @tylerfortune8.


### Changed
- Start using "changelog" over "change log" since it's the common usage.


### Removed
- Section about "changelog" vs "CHANGELOG".

## [0.3.0] - 2015-12-03
### Added
- RU translation from @aishek.


## [0.2.0] - 2015-10-06
### Changed
- Remove exclusionary mentions of "open source" since this project can
benefit both "open" and "closed" source projects equally.

## [0.1.0] - 2015-10-06
### Added
- Answer "Should you ever rewrite a change log?".

### Changed
- Improve argument against commit logs.
- Start following [SemVer](http://semver.org) properly.

## [0.0.8] - 2015-02-17
### Changed
- Update year to match in every README example.


### Fixed
- Fix typos in recent README changes.

## [0.0.7] - 2015-02-16
### Added
- Link, and make it obvious that date format is ISO 8601.

### Changed
- Clarified the section on "Is there a standard change log format?".

### Fixed
- Fix Markdown links to tag comparison URL with footnote-style links.

## [0.0.6] - 2014-12-12
### Added
- README section on "yanked" releases.

## [0.0.5] - 2014-08-09
### Added
- Markdown links to version tags on release headings.


## [0.0.4] - 2014-08-09
### Added
- Better explanation of the difference between the file ("CHANGELOG")
and its function "the change log".

### Changed
- Refer to a "change log" instead of a "CHANGELOG" throughout the site
to differentiate between the file and the purpose of the file — the
logging of changes.

### Removed
- Remove empty sections from CHANGELOG, they occupy too much space and
create too much noise in the file. People will have to assume that the
missing sections were intentionally left out because they contained no
notable changes.

## [0.0.3] - 2014-08-09
### Added
- "Why should I care?" section mentioning The Changelog podcast.

## [0.0.2] - 2014-07-10
### Added
- Explanation of the recommended reverse chronological release ordering.

src/PHP/Include/inc_Github.php
<?php
function SendToGithub(){
    // https://developer.github.com/v3/issues/#create-an-issue

    // test token access:
	//$repos = github_request('https://api.github.com/user/repos?page=1&per_page=100');
	//print_r($repos);
	//$events = github_request('https://api.github.com/users/:username/events?page=1&per_page=5');
	//$events = github_request('https://api.github.com/users/:username/events/public?page=1&per_page=5');
	//print_r($events);
	//$feeds = github_request('https://api.github.com/feeds/:username?page=1&per_page=5');
	//print_r($feeds);

	if(!isset($_POST['title']) && !isset($_POST['issue']))
	{
		die("Error, bad params.");
	}

	//TODO: you can add labels, milestone, assignees
	/*
	{
    "title": "Found a bug",
    "body": "I'm having a problem with this.",
    "assignees": [
    "octocat"
    ],
    "milestone": 1,
    "labels": [
    "bug"
    ]
    }	*/

    $appVersion = "?";

    if(isset($_GET['AppVersion'])) {$appVersion = $_GET['AppVersion'];}

	$named_array = array(
					  "title" => ('New Report '.$_POST['title'].' Ver '.$appVersion)
 					 ,"body" => ($_POST['issue'])

			);

	$data = createJsonBody($named_array);
    if($GLOBALS["DevMode"]) print_r($data);

	$newIssue = github_request(GetGithubIssueURL($GLOBALS['GithubUsername'], $GLOBALS['GithubRepository']), $data);
	if($GLOBALS["DevMode"]) print_r($newIssue);

} // end github

// Github REST API example
function github_request($url, $data){
    Global $GithubUsernameToken;

    $ch = curl_init();

    // Basic Authentication with token
    // https://developer.github.com/v3/auth/
    // https://github.com/blog/1509-personal-api-tokens
    // https://github.com/settings/tokens
    $access = $GithubUsernameToken;
    curl_setopt($ch, CURLOPT_USERPWD, $access);

    curl_setopt($ch, CURLOPT_URL, $url);

    curl_setopt($ch, CURLOPT_HTTPHEADER, array(
		 'Accept: application/vnd.github.v3+json'
		,'Content-Type: application/json')
		);

    curl_setopt($ch, CURLOPT_USERAGENT, 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.113 Safari/537.36');
    curl_setopt($ch, CURLOPT_HEADER, 0);

    curl_setopt($ch, CURLOPT_POSTFIELDS, $data);

    curl_setopt($ch, CURLOPT_TIMEOUT, 30);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);

    $output = curl_exec($ch);
    curl_close($ch);
    $result = json_decode(trim($output), true);
    return $result;
}

function GetGithubIssueURL($username, $repository){

    return ('https://api.github.com/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/issues');
}


function createJsonBody(array $parameters)
{
    return (count($parameters) === 0) ? null : json_encode($parameters, empty($parameters) ? JSON_FORCE_OBJECT : 0);
}
?>

src/PHP/recaptcha-v3-verify.php
<?php
/**
 * @copyright Copyright (c) 2015, Google Inc.
 * @link      https://www.google.com/recaptcha
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */

/*******************************************/
$DevMode = false;
/*******************************************/
session_start();
$_SESSION["IsRecaptchaOK"] = false;

require __DIR__ . '/appengine-https.php';

// Initiate the autoloader. The file should be generated by Composer.
// You will provide your own autoloader or require the files directly if you did
// not install via Composer.
require_once __DIR__ . '/Vendor/Recaptcha/autoload.php';

// Register API keys at https://www.google.com/recaptcha/admin
$siteKey = '';
$secret = '';

// update config.php it with your keys to run the examples or copy to
// config.development.php and set dev/private keys and set $DevMode = true.
if(!$DevMode){
    if ($siteKey == '' && is_readable(__DIR__ . '/config.php')) {
        $config = include __DIR__ . '/config.php';
        $siteKey = $config['v3']['site'];
        $secret = $config['v3']['secret'];
    }
}else{
    $config = include __DIR__ . '/config.development.php';
    $siteKey = $config['v3']['site'];
    $secret = $config['v3']['secret'];
}

// Effectively we're providing an API endpoint here that will accept the token, verify it, and return the action / score to the page
$recaptcha = new \ReCaptcha\ReCaptcha($secret);
$resp = $recaptcha->setExpectedHostname($_SERVER['SERVER_NAME'])
                  ->setExpectedAction('homepage')
                  ->setScoreThreshold(0.5)
                  ->verify($_GET['token'], $_SERVER['REMOTE_ADDR']);


if($resp->isSuccess()){
    if($resp->getScore() > 0.4) $_SESSION["IsRecaptchaOK"] = true;
}

//header('Content-type:application/json');
echo json_encode($resp->toArray());


src/PHP/recaptcha-v3-request-scores.php
<?php
/**
 * @copyright Copyright (c) 2015, Google Inc.
 * @link      https://www.google.com/recaptcha
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */

/*******************************************/
$DevMode = true;
/*******************************************/

require __DIR__ . '/appengine-https.php';

// Initiate the autoloader. The file should be generated by Composer.
// You will provide your own autoloader or require the files directly if you did
// not install via Composer.
require_once __DIR__ . '/Vendor/Recaptcha/autoload.php';

// Register API keys at https://www.google.com/recaptcha/admin
$siteKey = '';
$secret = '';

// Copy the config.php.dist file to config.php and update it with your keys to run the examples
// Copy the config.php.dist file to config.php and update it with your keys to run the examples
if(!$DevMode){
    if ($siteKey == '' && is_readable(__DIR__ . '/config.php')) {
        $config = include __DIR__ . '/config.php';
        $siteKey = $config['v3']['site'];
        $secret = $config['v3']['secret'];
    }
}else{
    $config = include __DIR__ . '/config.development.php';
    $siteKey = $config['v3']['site'];
    $secret = $config['v3']['secret'];
}

// reCAPTCHA supports 40+ languages listed here: https://developers.google.com/recaptcha/docs/language
$lang = 'en';


?>
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width,height=device-height,minimum-scale=1" />
    <link rel="shortcut icon" href="https://www.gstatic.com/recaptcha/admin/favicon.ico" type="image/x-icon" />
    <link rel="canonical" href="https://recaptcha-demo.appspot.com/recaptcha-v2-request-scores.php" />
    <script type="application/ld+json">{ "@context": "http://schema.org", "@type": "WebSite", "name": "reCAPTCHA demo - Request scores", "url": "https://recaptcha-demo.appspot.com/recaptcha-v2-request-scores.php" }</script>
    <meta name="description" content="reCAPTCHA demo - Request scores" />
    <meta property="og:url" content="https://recaptcha-demo.appspot.com/recaptcha-v2-request-scores.php" />
    <meta property="og:type" content="website" />
    <meta property="og:title" content="reCAPTCHA demo - Request scores" />
    <meta property="og:description" content="reCAPTCHA demo - Request scores" />
    <link rel="stylesheet" type="text/css" href="Css/examples.css" />
    <title>reCAPTCHA demo - Request scores</title>
</head>
<body>
    <header>
        <h1>reCAPTCHA demo</h1>
        <h2>Request scores</h2>
        <p>
            <a href="/">↤ Home</a>
        </p>
    </header>
    <main>
        <?php
        if ($siteKey === '' || $secret === ''):
        ?>
        <h2>Add your keys</h2>
        <p>
            If you do not have keys already then visit
            <kbd>
                <a href="https://www.google.com/recaptcha/admin">https://www.google.com/recaptcha/admin</a>
            </kbd> to generate them. Edit this file and set the respective keys in
            <kbd>$siteKey</kbd> and
            <kbd>$secret</kbd>. Reload the page after this.
        </p>
        <?php
        else:
            // Add the g-recaptcha tag to the form you want to include the reCAPTCHA element
        ?>
        <p>reCAPTCHA will provide a score for this request.</p>
        <ol id="recaptcha-steps">
            <li class="step0">reCAPTCHA script loading</li>
            <li style="display:none" class="step1">
                <kbd>grecaptcha.ready()</kbd> fired, calling
                <pre>
                    grecaptcha.execute('<?php echo $siteKey; ?>', {action: 'homepage'})'
                </pre>
            </li>
            <li style="display:none" class="step2">
                Received token from reCAPTCHA service, sending to our backend with
                <kbd>
                    fetch('/recaptcha-v3-verify.php?token='+
                    <span class="token">123</span>)
                </kbd>
            </li>
            <li style="display:none" class="step3">
                Received response from our backend:
                <pre class="response">response</pre>
            </li>
        </ol>
        <p>
            <a href="/recaptcha-v3-request-scores.php">⟳ Try again</a>
        </p>

        <script src="https://www.google.com/recaptcha/api.js?render=<?php echo $siteKey; ?>"></script>
        <script>
    const steps = document.getElementById('recaptcha-steps');
    grecaptcha.ready(function() {
        document.querySelector('.step1').style.display = 'list-item';
        grecaptcha.execute('<?php echo $siteKey; ?>', {action: 'homepage'}).then(function(token) {
            document.querySelector('.token').innerHTML = token;
            document.querySelector('.step2').style.display = 'list-item';

            fetch('/recaptcha-v3-verify.php?token='+token).then(function(response) {
                response.json().then(function(data) {
                    document.querySelector('.response').innerHTML = JSON.stringify(data, null, 2);
                    document.querySelector('.step3').style.display = 'list-item';
                });
            });
        });
    });
        </script>
        <?php
        endif;?>
    </main>
</body>
</html>


src/PHP/index.php
<?php

/*******************************************/
$DevMode = false;

/* github */
$GithubUsernameToken = "";
$GithubUsername = "JavierCanon";
$GithubRepository = "web-github-issue-reporting";

/*******************************************/
session_start();

if(!isset($_POST['title']) && !isset($_POST['issue']) && !isset($_POST['submit'])){
    // start session var.
    $_SESSION["IsRecaptchaOK"] = false;
}

require __DIR__ . '/appengine-https.php';

// Initiate the autoloader. The file should be generated by Composer.
// You will provide your own autoloader or require the files directly if you did
// not install via Composer.
require_once __DIR__ . '/Vendor/Recaptcha/autoload.php';

// Register API keys at https://www.google.com/recaptcha/admin
$siteKey = '';
$secret = '';

// update config.php it with your keys to run the examples or copy to
// config.development.php and set dev/private keys and set $DevMode = true.
if(!$DevMode){
    if ($siteKey == '' && is_readable(__DIR__ . '/config.php')) {
        $config = include __DIR__ . '/config.php';
        $siteKey = $config['v3']['site'];
        $secret = $config['v3']['secret'];
    }
}else{
    $config = include __DIR__ . '/config.development.php';
    $siteKey = $config['v3']['site'];
    $secret = $config['v3']['secret'];
}

// reCAPTCHA supports 40+ languages listed here: https://developers.google.com/recaptcha/docs/language
$lang = 'en';

include_once("Include/inc_DB.php");
include_once("Include/inc_Github.php");

$showMsg = "";
if(isset($_POST['title']) && isset($_POST['issue']) && isset($_POST['submit'])){

    if($_SESSION["IsRecaptchaOK"] == true ){
        SendToGithub();
        $showMsg ='<div class="alert alert-success">Thanks, report sended.</div>';
    } else{
        $showMsg ='<div class="alert alert-danger">Error sending the report.</div>';

    }
}


?>
<!doctype html>
<html lang="en">
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
    <meta name="description" content="" />
    <meta name="author" content="Javier Cañon" />
    <link rel="icon" href="favicon.ico" />

    <title>Issue Reporting</title>

    <!-- Bootstrap core CSS -->
    <link crossorigin='anonymous' href='https://maxcdn.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css' rel='stylesheet' />
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous" />

    <!-- Custom styles for this template -->
    <link href="Css/StyleSheet.css" rel="stylesheet" />
</head>

<body>
    <?php
    if ($siteKey === '' || $secret === ''):
    ?>
    <h2>Add your keys</h2>
    <p>
        If you do not have keys already then visit
        <kbd>
            <a href="https://www.google.com/recaptcha/admin">https://www.google.com/recaptcha/admin</a>
        </kbd> to generate them. Edit this file and set the respective keys in
        <kbd>$siteKey</kbd> and
        <kbd>$secret</kbd>. Reload the page after this.
    </p>
    <?php
    else:
        // Add the g-recaptcha tag to the form you want to include the reCAPTCHA element
    ?>
    <form id="formReport" action="index.php" class="form-reporting" method="post" enctype="multipart/form-data">
        <div class="text-center mb-4">

            <h1 class="h3 mb-3 font-weight-normal">
                <i class="fas fa-bug fa-2x"></i> Report an Issue
            </h1>
            <p>
                Please use the next form to report an issue, with the details to reproduce and links to cloud files, photos or videos
                (Google Drive, Google Photos, Microsoft Onedrive, Dropbox, Wetransfer, etc.).
            </p>
        </div>

        <?php echo $showMsg; ?>

        <div class="form-label-group" id="honeybotpot">
            <input type="email" class="form-control" name="email" id="email" value="" placeholder="E-mail" />
        </div>

        <div class="form-label-group">
            <label for="title">Issue Title</label>
            <input type="text" id="title" name="title" class="form-control" placeholder="Issue Title" required autofocus
                value="<?php if(isset($_GET['title'])) { echo $_GET['title']  ;} ?>" />
        </div>

        <div class="form-label-group">
            <label for="issue">Issue Text</label>
            <textarea id="issue" name="issue" class="form-control" placeholder="Issue Text" maxlength="4000"
                rows="10" required>
                <?php if(isset($_GET['issue'])) { echo $_GET['issue'] ;}?>
            </textarea>
        </div>

        <button class="btn btn-lg btn-primary btn-block" type="submit" name="submit">
            <i class="fa fa-share-square" aria-hidden="true"></i> Submit
        </button>

        <div class="alert alert-danger" id="sendMsgError" style="display:none;">
            <strong>Error</strong> Cant sent the report.
        </div>

        <div class="alert alert-success" id="sendMsgOK" style="display:none;">
            <strong>Thanks,</strong> the report was sended.
        </div>
        <?php if($DevMode){ ?>
        <div id="response" class="alert alert-danger"></div><?php } ?>
        <p class="mt-5 mb-3 text-muted text-center">&copy; 2019</p>
    </form>

    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.6/umd/popper.min.js" integrity="sha384-wHAiFfRlMFy6i5SRaxvfOCifBUQy1xHdJ/yoi7FRNXMRBu5WHdZYu1hA6ZOblgut" crossorigin="anonymous"></script>
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/js/bootstrap.min.js" integrity="sha384-B0UglyR+jN6CkvvICOB2joaf5I4l3gm9GU6Hc1og6Ls7i6U/mkkaduKaBhlAXv9k" crossorigin="anonymous"></script>
    <script src="https://www.google.com/recaptcha/api.js?render=<?php echo $siteKey; ?>"></script>


    <script>

        $.fn.captcha = function (param) {

            var params = $.extend({
                idCaptchaText: 'captchaText',
                idCaptchaInput: 'captchaInput',
                class: ''
            }, param);

            var submit = $(this).find('button[type=submit]');
            submit.attr('disabled', 'disabled');
            submit.addClass('btn-secondary disabled');

            $('<div class="form-label-group"><label id="' + params.idCaptchaText + '"></label>' + '<input class="form-control" placeholder="Anti SPAM" id="' + params.idCaptchaInput + '" aria-label="Captcha Input" type="text" required="required" /></div>').insertBefore(submit);

            var label = this.find('#' + params.idCaptchaText);
            var input = this.find('#' + params.idCaptchaInput);

            var rndmNr1 = Math.floor(Math.random() * 10),
                rndmNr2 = Math.floor(Math.random() * 10),
                totalNr = rndmNr1 + rndmNr2;

            $(label).text('How much is? ' + rndmNr1 + ' + ' + rndmNr2 + ' =');

            $(input).keyup(function () {
                if ($(this).val() == totalNr)
                    submit.removeAttr('disabled').removeClass('btn-secondary disabled').addClass('btn-primary');
                else
                    submit.attr('disabled', 'disabled');
            });

            return this;
        }


        $(document).ready(function () {
            $('#formReport').captcha();
            $('#honeybotpot').hide();
        });

        var frm = $('#formReport');

        frm.submit(function (e) {

            //e.preventDefault();

            // check honeypot
            if (document.getElementById("email").value) {
                console.log('An error occurred.');
                return false;
            }

            if (!reCaptchaSuccess) { $('#sendMsgError').show(); return false; }
            if (reCaptchaScore < 0.4) { $('#sendMsgError').show(); return false; }

            return true;
        });

    </script>

    <script>

     var reCaptchaScore = 0;
     var reCaptchaSuccess = false;

     grecaptcha.ready(function() {

        grecaptcha.execute('<?php echo $siteKey; ?>', {action: 'homepage'}).then(function(token) {
            fetch('/recaptcha-v3-verify.php?token='+token).then(function(response) {
                response.json().then(function(data) {

                   var resp = JSON.parse(JSON.stringify(data));
                   reCaptchaScore = resp.score;
                   reCaptchaSuccess = resp.success;
                   <?php if($DevMode) { ?> document.querySelector('#response').innerHTML = JSON.stringify(data, null, 2); <?php } ?>
                });
            });
        });
    });
    </script>
    <?php
    endif;?>
</body>
</html>
